### PR TITLE
Fetch tiles from s3

### PR DESCRIPTION
**Pull request by jczaplew**
Modifies `HGT` to fetch data from S3 instead of the local file system. I tested this approach on an EC2 instance in the same region as the data and it takes approximately 300ms to fetch a tile. 

Many instances of `setImmediate` are also removed because we no longer block with calls to `fs`.

Instead of modifying the tile loader in-place I considered adding S3 as a separate tile loader, but it seemed like it would add more complexity than it is worth

### CODE REVIEW

File: hgt/index.js [hunk 1]
```diff
@@ -1,39 +1,31 @@
-const fs = require('fs');
+const { S3Client, GetObjectCommand } = require('@aws-sdk/client-s3')
 const THREE_ARC_SECOND = 1442401 * 2;
 const ONE_ARC_SECOND = 12967201 * 2;
 
+const s3Client = new S3Client({region: process.env.AWS_REGION || 'us-east-1'});
```
Comment (jczaplew) [inline hgt/index.js:5]: Based on [this example](https://github.com/awsdocs/aws-doc-sdk-examples/blob/master/javascriptv3/example_code/s3/src/s3_getobject.ts#L34), I'm reasonably certain we don't need to explicitly pass credentials to the client if running on ECS

File: hgt/index.js [hunk 2]
```diff
@@ -1,39 +1,31 @@
...
-                                resolution: resAndSize.resolution,
-                                size: resAndSize.size,
-                                options,
-                                swLngLat,
-                            });
-                        });
-                    });
-                });
-            });
+        const buffer = await streamToBuffer(dem.Body)
```
Comment (mcglincy) [inline hgt/index.js]: very small nit: missing trailing ;

File: hgt/index.js [hunk 3]
```diff
@@ -1,39 +1,31 @@
...
+        return callback(undefined, {
+            buffer,
+            resolution: resAndSize.resolution,
+            size: resAndSize.size,
+            options,
+            swLngLat,
         });
+    }).catch((error) => {
+        console.log(`Error fetching tile ${path}`, error)
+        callback(`Error fetching tile ${path}`)
```
Comment (mcglincy) [inline hgt/index.js]: double-checking: can the callback handle passing the error arg as just a string? I see we pass the caught error in other callback() invocations.

**Review by mcglincy — APPROVED** [review]
One nit, one double-check question. LGTM.

File: hgt/index.js (hunk 3 above)
Comment (jczaplew) [reply hgt/index.js]: Indeed. The type of the callback arg doesn't matter so much as the presence of it

**Comment by jczaplew** [comment]
Tested this by manually pushing an image to ECR, creating a new task that uses that image, and manually running the task on  an instance that I have ssh access to. I then verified that the service works as expected with curl.

**Pull request merged**